fgpyo/sam/builder.py

```diff
@@ -444,8 +380,16 @@ def add_pair(
             raise ValueError("Cannot use chrom in combination with chrom1 or chrom2")
 
         chrom = sam.NO_REF_NAME if chrom is None else chrom
-        chrom1 = next(c for c in (chrom1, chrom) if c is not None)
-        chrom2 = next(c for c in (chrom2, chrom) if c is not None)
+
+        if start1 != sam.NO_REF_POS:
+            chrom1 = next(c for c in (chrom1, chrom) if c is not None)
+        else:
+            chrom1 = sam.NO_REF_NAME
+
+        if start2 != sam.NO_REF_POS:
+            chrom2 = next(c for c in (chrom2, chrom) if c is not None)
+        else:
+            chrom2 = sam.NO_REF_NAME
```

Comment:
Without these edge cases, `SamBuilder` would build partially incorrect pairs with syntax like:

```python
r1, r2 = builder.add_pair(chrom="chr1", start1=100)
```

In the above, R1 would be mapped to `chr1:100`.

R2 would be unmapped, but would still have its reference sequence name set to `chr1` which is non-sensical.